src/lib/registration/widgets/registration_code.rb

```diff
@@ -73,13 +74,38 @@ def register
         Storage::InstallationOptions.instance.base_registered = true
       end
 
+      def validate
+        return error(_("System already registered.")) if Registration.is_registered?
```

Comment:
Yep, I will just log in case of already registered, but also in that case we could show the base product label instead of the widget as already registered in the content of the casp overview client.